types/test.d.ts

```diff
@@ -0,0 +1,203 @@
+  isStrictly: Assertions.Equal
+
+  notEqual: Assertions.NotEqual
+  notEquals: Assertions.NotEqual
+  inequal: Assertions.NotEqual
+  notStrictEqual: Assertions.NotEqual
+  notStrictEquals: Assertions.NotEqual
+  isNotEqual: Assertions.NotEqual
+  isNot: Assertions.NotEqual
+  doesNotEqual: Assertions.NotEqual
+  isInequal: Assertions.NotEqual
+
+  same: Assertions.Equal
+  equivalent: Assertions.Equal
+  looseEqual: Assertions.Equal
+  looseEquals: Assertions.Equal
+  deepEqual: Assertions.Equal
+  deepEquals: Assertions.Equal
+  isLoose: Assertions.Equal
+  looseIs: Assertions.Equal
+
+  notSame: Assertions.NotEqual
+  inequivalent: Assertions.NotEqual
+  looseInequal: Assertions.NotEqual
+  notDeep: Assertions.NotEqual
+  deepInequal: Assertions.NotEqual
+  notLoose: Assertions.NotEqual
+  looseNot: Assertions.NotEqual
+
+  strictSame: Assertions.Equal
+  strictEquivalent: Assertions.Equal
+  strictDeepEqual: Assertions.Equal
+  sameStrict: Assertions.Equal
+  deepIs: Assertions.Equal
+  isDeeply: Assertions.Equal
+  isDeep: Assertions.Equal
+  strictDeepEquals: Assertions.Equal
+
+  strictNotSame: Assertions.NotEqual
+  strictInequivalent: Assertions.NotEqual
+  strictDeepInequal: Assertions.NotEqual
+  notSameStrict: Assertions.NotEqual
+  deepNot: Assertions.NotEqual
+  notDeeply: Assertions.NotEqual
+  strictDeepInequals: Assertions.NotEqual
+  notStrictSame: Assertions.NotEqual
+
+  match: Assertions.Match
+  has: Assertions.Match
+  hasFields: Assertions.Match
+  matches: Assertions.Match
+  similar: Assertions.Match
+  like: Assertions.Match
+  isLike: Assertions.Match
+  includes: Assertions.Match
+  include: Assertions.Match
+  contains: Assertions.Match
+
+  notMatch: Assertions.Match
+  dissimilar: Assertions.Match
+  unsimilar: Assertions.Match
+  notSimilar: Assertions.Match
+  unlike: Assertions.Match
+  isUnlike: Assertions.Match
+  notLike: Assertions.Match
+  isNotLike: Assertions.Match
+  doesNotHave: Assertions.Match
+  isNotSimilar: Assertions.Match
+  isDissimilar: Assertions.Match
+
+  type: Assertions.Type
+  isa: Assertions.Type
+  isA: Assertions.Type
+}
+
+export declare namespace Options {
+  export interface Bag {
+    [key: string]: any
+  }
+
+  export interface Pragma {
+    [key: string]: boolean
+  }
+
+  export interface Assert extends Bag {
+    todo?: boolean | string,
+    skip?: boolean | string,
+    diagnostic?: boolean
+  }
+
+  export interface Spawn extends Assert {
+    bail?: boolean,
+    timeout?: number
+  }
+
+  export interface Test extends Assert {
+    timeout?: number,
+    bail?: boolean,
+    autoend?: boolean,
+    buffered?: boolean,
+    jobs?: number,
+    grep?: RegExp[],
+    only?: boolean,
+    runOnly?: boolean
+  }
+}
+
+export declare namespace Assertions {
+  export type Basic = (obj: any, message?: string, extra?: Options.Assert) => boolean
+
+  export type Throws = (fn?: (a: any) => any, expectedError?: Error, message?: string, extra?: Options.Assert) => boolean
```

Comment:
I think the following is more accurate to how `throws` can be used. I had to substitute `Error` for `Object` because of the example given in the documentation.
```suggestion
  type AnyFunction = (...args: any[]) => any
  export interface Throws {
    (fn: AnyFunction): boolean;
    (fn: AnyFunction, message: string): boolean;
    (fn: AnyFunction, expectedError: Object): boolean;
    (fn: AnyFunction, expectedError: Object, message: string): boolean;
    (fn: AnyFunction, expectedError: Object, extra: Options.Assert): boolean
    (fn: AnyFunction, message: string, extra: Options.Assert): boolean
    (fn: AnyFunction, expectedError: Object, message: string, extra: Options.Assert): boolean
    (fn: AnyFunction, arg1?: string | Object, arg2?: string | Object | Options.Assert, arg3?: Options.Assert): boolean
  }
```